source/MRVoxels/MRFloatGrid.cpp

```diff
@@ -45,16 +42,26 @@ FloatGrid resampled( const FloatGrid& grid, const Vector3f& voxelScale, Progress
 
     ProgressInterrupter interrupter( dummyProgressCb );
     // openvdb::util::NullInterrupter template argument to avoid tbb inconsistency
-    openvdb::tools::resampleToMatch<openvdb::tools::BoxSampler, openvdb::util::NullInterrupter>( grid_, *dest, interrupter );
 
-    // restore original grid class
-    if ( backupClass == openvdb::GRID_LEVEL_SET )
-        const_cast< openvdb::FloatGrid& >( grid_ ).setGridClass( openvdb::GRID_LEVEL_SET );
+    bool failed = true;
+    if ( dest->getGridClass() == openvdb::GRID_LEVEL_SET )
+    {
+        try {
+            dest = openvdb::tools::doLevelSetRebuild( grid_, 0.f, 1, 1, &dest->constTransform(), &interrupter );
+            failed = false;
+        }
+        catch( ... )
+        {}
```

Comment:
please add a comment what exceptions do we expected here and why they are ignored